sn_networking/src/lib.rs

```diff
@@ -385,6 +385,13 @@ impl Network {
             let res = self
                 .put_record_once(record.clone(), verify_store.clone())
                 .await;
+
+            if matches!(res, Err(Error::RecordNotEnoughCopies(_)))
+                || matches!(res, Err(Error::RecordNotFound))
+            {
+                continue;
+            }
```

Comment:
I don't think we propagate the `RecordNotEnoughCopies` uptill here? It is intercepted and modified during `get_record`.

Also should we perform `retries+1` here to prevent infinte loop?